src/NLog/Config/NLogXmlElement.cs

```diff
@@ -212,19 +221,59 @@ public void AssertName(params string[] allowedNames)
             throw new InvalidOperationException("Assertion failed. Expected element name '" + string.Join("|", allowedNames) + "', actual: '" + this.LocalName + "'.");
         }
 
+        /// <summary>
+        /// Try to get all errors occured during xml parsing.
+        /// A return value indicates whether there any error.
+        /// </summary>
+        /// <param name="errors">Array of error messages</param>
+        public bool TryGetParsingErrors(out string[] errors)
```

Comment:
I don't prefer `out` parameters. Could be change this to `public string GetParsingErrors()`? 

The "try" pattern is nice, but IMO more suitable on parsing.